nostr/relay_manager.py

```diff
@@ -14,43 +18,78 @@ class RelayException(Exception):
 
 
 
+@dataclass
 class RelayManager:
-    def __init__(self) -> None:
+    def __post_init__(self):
         self.relays: dict[str, Relay] = {}
-        self.message_pool = MessagePool()
+        self.message_pool: MessagePool = MessagePool()
+        self.lock: Lock = Lock()
 
-    def add_relay(self, url: str, read: bool=True, write: bool=True, subscriptions={}):
-        policy = RelayPolicy(read, write)
-        relay = Relay(url, policy, self.message_pool, subscriptions)
-        self.relays[url] = relay
+    def add_relay(
+            self, 
+            url: str, 
+            policy: RelayPolicy = RelayPolicy(),
+            ssl_options: dict = None,
+            proxy_config: RelayProxyConnectionConfig = None):
```

Comment:
Sorry I'm late on this one.

This default value on `proxy_config` overrides the default non-None value on the `Relay` class, so you end up with:

```
relay.proxy_config == None
```

Which then fails when it tries to access `proxy_config` in `relay.connect()`.

I don't like the idea of requiring an all-`None` `RelayProxyConnectionConfig` in order to successfully call `relay.connect()`. The prior form of `self.proxy_config.host if self.proxy_config is not None else None` is still preferable to me, despite its verbosity.

Reply:
Suggested fix in #62. Along with one other minor bugfix.